caffe2/python/modeling/initializers.py

```diff
@@ -29,6 +29,33 @@ def create_param(self, param_name, init_net, shape):
             shape=shape,
         )
 
+class pFP16Initializer(object):
+    def __init__(self, operator_name=None, **kwargs):
+        self.operator_name = operator_name
+        self.operator_kwargs = kwargs
+
+    def update(self, operator_name, kwargs):
+        if self.operator_name is not None:
+            raise Exception("Operator name overwrites are not allowed")
+        self.operator_name = operator_name
+        self.operator_kwargs = kwargs
+
+    def create_param(self, param_name, init_net, shape):
```

Comment:
Because right now it'll look like:

```
param_fp32_info = super(Initializer, self).create_param(
        param_name + "fp32",
        init_net,
        shape)

param = init_net.FloatToHalf(param_fp32_info.blob,
                                                param_name)

return ParameterInfo(
        param_id=None,
        param=param,
        shape=shape,
        blob_copy={'float' : param_fp32_info.blob})
```

That's not really cleaner in my opinion